File: mqtt/src/main/java/org/sheedon/mqtt/Response.java
package org.sheedon.mqtt;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * 反馈响应类
 *
 * code + 描述 + 内容
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/2/11 12:48
 */
public class Response{
    final int code;
    final String message;
    final @Nullable ResponseBody body;

    protected Response(ResponseBuilder builder) {
        this.code = builder.code;
        this.message = builder.message;
        this.body = builder.body;
    }

    public int code() {
        return code;
    }

    public boolean isSuccessful() {
        return code >= 200 && code < 300;
    }

    public String message() {
        return message;
    }

    public ResponseBody body() {
        return body;
    }


    public ResponseBuilder newBuilder() {
        return new ResponseBuilder(this);
    }

    @NonNull
    @Override
    public String toString() {
        return "Response{code="
                + code
                + ", message="
                + message
                + ", body="
                + (body != null ? body.getBody() : "null")
                + '}';
    }
}

File: mqtt/src/main/java/org/sheedon/mqtt/Response.kt
package org.sheedon.mqtt

import org.sheedon.rr.dispatcher.model.BaseResponse

/**
 * Basic feedback class, the content to be included includes "feedback subject" and "feedback message body"
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/30 11:01 上午
 */
class Response(
    backTopic: String? = "",
    message: String? = "",
    body: ResponseBody? = null
) : BaseResponse<String, ResponseBody>(backTopic, message, body) {

    override fun backTopic(): String {
        return super.backTopic() ?: ""
    }

    override fun message(): String {
        return super.message() ?: ""
    }

}